foursight_core/react/api/cognito.py

```diff
@@ -0,0 +1,364 @@
+# AWS Cognito authentication support functions.
+
+import jwt as jwtlib
+from jwt import PyJWKClient
+import requests
+import os
+from typing import Tuple
+from foursight_core.react.api.encoding_utils import base64_encode
+from foursight_core.react.api.envs import Envs
+from foursight_core.react.api.gac import Gac
+from foursight_core.react.api.jwt_utils import jwt_encode
+from foursight_core.react.api.misc_utils import get_request_domain, get_request_origin, memoize
+
+
+COGNITO_BASE_URL = "https://cognito-idp.us-east-1.amazonaws.com"
+
+
+def get_cognito_oauth_config(request: dict) -> dict:
+    """
+    Returns all necessary configuration info for our AWS Cognito authentication server.
+    Retrieved via either environment variables of AWS Secrets Manager.
+    :returns: Dictionary with AWS Cognito configuration info.
+    """
+    config = _get_cognito_oauth_config_basic()
+    config["callback"] = os.environ.get("FOURSIGHT_COGNITO_CALLBACK",
```

Comment:
Perspicacious observation. Yes, actually, this is useful for running locally in (what I call) cross-origin mode, where the React UI is running on its own on localhost:3000 (via npm) and the Foursight app is running locally on localhost:8000 (so that live React code updates can be made and seen immediately on localhost:3000). In this case we need to set FOURSIGHT_COGNITO_CALLBACK to localhost:3000, rather than to what the default would have been, localhost:8000.